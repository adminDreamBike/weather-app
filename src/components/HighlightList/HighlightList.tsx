"use client";

import { Flex, Text } from "@chakra-ui/react";
import { Highlight } from "@/components/Highlight/Highlight";

export const HighlightList = ({ data }) => {
  const { current } = data || {};

  return (
    <>
      <Text color="#e4e4e8">Todays Highglights</Text>
      <Flex gap="20px">
        <Highlight
          title="Wind Status"
          value={current?.wind_speed}
          units="mph"
          wind_deg={current.wind_deg}
        />
        <Highlight
          title="Humidity"
          value={current?.humidity}
          units="%"
          progress={current?.humidity}
        />
        <Highlight
          title="Visibility"
          value={current?.visibility}
          units="miles"
        />
        <Highlight title="Air Pressure" value={current?.pressure} units="mb" />
      </Flex>
    </>
  );
};
