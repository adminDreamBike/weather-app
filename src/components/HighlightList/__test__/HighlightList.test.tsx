import { render } from "@testing-library/react";
import { HighlightList } from "../HighlightList";
import React from "react";
import { ChakraUIProvider } from "@/providers/ChakraUIProvider";

const currentValues = {
  clouds: 75,
  dew_point: 293.92,
  dt: 1729633812,
  feels_like: 294.81,
  humidity: 99,
  pressure: 1013,
  sunrise: 1729594088,
  sunset: 1729637096,
  temp: 294.08,
  uvi: 0.52,
  visibility: 10000,
  weather: [
    {
      description: "broken clouds",
      icon: "04d",
      id: 803,
      main: "Clouds",
    },
  ],
  wind_deg: 50,
  wind_gust: 0,
  wind_speed: 3.58,
};
const dailyValues = {
  clouds: 100,
  dew_point: 288.68,
  dt: 1729612800,
  feels_like: { day: 291.04, night: 288.92, eve: 294.81, morn: 290.39 },
  humidity: 99,
  moon_phase: 0.7,
  moonrise: 1729655700,
  moonset: 1729612320,
  pop: 1,
  pressure: 1017,
  rain: 21.24,
  summary: "Expect a day of partly cloudy with rain",
  sunrise: 1729594088,
  sunset: 1729637096,
  temp: { day: 290.65, min: 288.75, max: 294.08, night: 288.75, eve: 294.08 },
  uvi: 5.72,
  weather: [
    {
      description: "broken clouds",
      icon: "04d",
      id: 803,
      main: "Clouds",
    },
  ],
  wind_deg: 88,
  wind_gust: 1.56,
  wind_speed: 1.07,
};
const minutelyValues = {
  dt: 1729633860,
  precipitation: 0,
};
const defaultProps = {
  current: { ...currentValues },
  daily: { ...dailyValues },
  hourly: [{ ...currentValues }],
  lat: 6.2443,
  lon: -75.5736,
  minutely: [{ ...minutelyValues }],
  timezone: "America/Bogota",
  timezone_offset: -18000,
};
describe("", () => {
  test("", () => {
    const { asFragment } = render(
      <ChakraUIProvider>
        <HighlightList data={defaultProps} />
      </ChakraUIProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
