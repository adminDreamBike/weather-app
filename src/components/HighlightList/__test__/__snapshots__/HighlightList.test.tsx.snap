// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[` 1`] = `
<DocumentFragment>
  <p
    class="chakra-text css-1infxn0"
  >
    Todays Highglights
  </p>
  <div
    class="css-kl5uk3"
  >
    <div
      class="chakra-card css-rrzei3"
    >
      <div
        class="chakra-card__body css-rzdfkz"
      >
        <p
          class="chakra-text css-15bto3y"
        >
          Wind Status
        </p>
        <div
          class="css-skjq34"
        >
          <p
            class="chakra-text css-1bsg0uq"
          >
            3.58
          </p>
          <p
            class="chakra-text css-1v15ftq"
          >
            mph
          </p>
        </div>
        <svg
          class="chakra-icon css-13otjrl"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          stroke="currentColor"
          stroke-width="0"
          style="transform: rotate(50deg);"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M429.6 92.1c4.9-11.9 2.1-25.6-7-34.7s-22.8-11.9-34.7-7l-352 144c-14.2 5.8-22.2 20.8-19.3 35.8s16.1 25.8 31.4 25.8H224V432c0 15.3 10.8 28.4 25.8 31.4s30-5.1 35.8-19.3l144-352z"
          />
        </svg>
      </div>
    </div>
    <div
      class="chakra-card css-rrzei3"
    >
      <div
        class="chakra-card__body css-rzdfkz"
      >
        <p
          class="chakra-text css-15bto3y"
        >
          Humidity
        </p>
        <div
          class="css-skjq34"
        >
          <p
            class="chakra-text css-1bsg0uq"
          >
            99
          </p>
          <p
            class="chakra-text css-1v15ftq"
          >
            %
          </p>
        </div>
        <div
          class="css-1i7xuna"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="99"
            class="css-1jrtelv"
            role="progressbar"
            style="width: 99%;"
          />
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-rrzei3"
    >
      <div
        class="chakra-card__body css-rzdfkz"
      >
        <p
          class="chakra-text css-15bto3y"
        >
          Visibility
        </p>
        <div
          class="css-skjq34"
        >
          <p
            class="chakra-text css-1bsg0uq"
          >
            10000
          </p>
          <p
            class="chakra-text css-1v15ftq"
          >
            miles
          </p>
        </div>
      </div>
    </div>
    <div
      class="chakra-card css-rrzei3"
    >
      <div
        class="chakra-card__body css-rzdfkz"
      >
        <p
          class="chakra-text css-15bto3y"
        >
          Air Pressure
        </p>
        <div
          class="css-skjq34"
        >
          <p
            class="chakra-text css-1bsg0uq"
          >
            1013
          </p>
          <p
            class="chakra-text css-1v15ftq"
          >
            mb
          </p>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
